backend/webserver/wittymail_server/emailapi_broker.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import openpyxl
+import emailapi
+import re
+import util.logger as logger
+
+log = logger.get_logger(__name__)
+
+# Names of the fodder to be used to create the emails
+email_fodder_names = []
+# Fodder content used to create the emails
+email_fodder = []
+
+# Index of the emailid in the email_fodder_names[]. The emailid will be used in the 'to' field
+EMAIL_FODDER_TO_INDEX = 0 
+# Index of the emailid in the email_fodder_names[]. The emailid will be used in the 'cc' field
+EMAIL_FODDER_CC_INDEX = 0
+
+# Extended fodder names that will be appended to the fodder names provided by the user (i.e. email_fodder_names)
+extended_email_fodder_names = ['email_subject', 'email_body', 'email_attachment', 'email_sent']
+extended_email_fodder_names_EMAIL_BODY_INDEX = -3
+extended_email_fodder_names_EMAIL_SUBJECT_INDEX = -4
+extended_email_fodder_names_EMAIL_ATTACHMENT_INDEX = -2
+
+# Extended fodder that will be appended to the fodder provided by the user (i.e. email_fodder[])
+extended_default_email_fodder = [None, None, None, False]
+
+# TODO Add check to detect extention type
+def save_fodder_from_file(loc, email_fodder_names_template = None):
+    global email_fodder
+    global email_fodder_names
+   
+    _email_fodder = []  
+
+    # TODO Check if we should read wb_obj.active or wb_obj[0]
+    wb_obj = openpyxl.load_workbook(loc)   
+    sheet_obj = wb_obj.active
+
+    for i in range(1, sheet_obj.max_row + 1):
+        cells = []
+
+        for j in range(1, sheet_obj.max_column + 1):
+            cells.append(str(sheet_obj.cell(row = i, column = j).value).encode('UTF8'))
+        _email_fodder.append(cells)
+
+    assert email_fodder_names_template == None or email_fodder_names == email_fodder_names_template, \
+           "**** The headers in the excel sheet seems to be invalid ****"
+    assert len(_email_fodder) > 1,            \
+           "**** There are no entries in the excel sheet! ****"
+
+    email_fodder_names = _email_fodder[0]
+    email_fodder_names.extend(extended_email_fodder_names)
+
+    email_fodder = _email_fodder[1:]
+
+    for r in email_fodder:
+        extended_email_fodder = list(extended_default_email_fodder)
+        # TODO Add template support for attachment and support different filetypes
+        extended_email_fodder[2] = r[1] + ".pdf"
+        r.extend(extended_email_fodder)
+
+def get_email_fodder_names():
+    return email_fodder_names
+
+def get_email_fodder():
+    return email_fodder
+
+def _template_to_str(_st, l):
```

Comment:
Can you add a comment here on what we expect to be in the template and some examples? I've already forgotten!